src/JointTrajectoryControllerBase.cpp

```diff
@@ -349,6 +374,25 @@ void JointTrajectoryControllerBase::goalCallback(GoalHandle goalHandle)
   newContext->mTrajectory = trajectory;
   newContext->mGoalHandle = goalHandle;
 
+  // Evaluate the trajectory at the current time.
+  auto offsetDesiredState = mControlledSpace->createState();
```

Comment:
I think the names `offsetDesiredState`, `offsetDesiredPosition`, and `offsetActualPosition` are misleading here. These seem to refer to `initialTrajectoryState`, `initialTrajectoryPosition`, and `actualSkeletonPosition`. None of these are really "offsets". You are doing  `initialTrajectoryPosition - actualSkeletonPosition` to _get_ the offset.